lib/modules/story_editor/screens/story_editor.dart

```diff
@@ -0,0 +1,111 @@
+import 'package:andax/models/story.dart';
+import 'package:andax/models/translation.dart';
+import 'package:andax/models/translation_asset.dart';
+import 'package:andax/modules/story_editor/screens/actors_editor.dart';
+import 'package:andax/modules/story_editor/screens/info_editor.dart';
+import 'package:andax/modules/story_editor/screens/narrative_editor.dart';
+import 'package:andax/shared/widgets/danger_dialog.dart';
+import 'package:flutter/material.dart';
+import 'package:provider/provider.dart';
+import 'package:uuid/uuid.dart';
+
+class StoryEditorScreen extends StatefulWidget {
+  const StoryEditorScreen({
+    this.story,
+    this.translation,
+    this.info,
+    Key? key,
+  }) : super(key: key);
+
+  final Story? story;
+  final Translation? translation;
+  final StoryInfo? info;
+
+  @override
+  StoryEditorState createState() => StoryEditorState();
+}
+
+class StoryEditorState extends State<StoryEditorScreen> {
+  final uuid = const Uuid();
+
+  late StoryInfo? info = widget.info;
+  late final Story story = widget.story ??
+      Story(
+        nodes: [],
+        startNodeId: '',
+        actors: [],
+      );
+  late final Translation translation = widget.translation ??
+      Translation(
+        language: '',
+        assets: {
+          'story': StoryTranslation(
+            id: '',
+            title: 'New story',
+          )
+        },
+      );
+
+  var _page = 0;
+  final _paging = PageController();
+
+  @override
+  void setState(void Function() fn) {
+    super.setState(fn);
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return Provider.value(
+      value: this,
+      child: WillPopScope(
+        onWillPop: () => showDangerDialog(
+          context,
+          'Leave editor? Unsaved progress will be lost!',
```

Comment:
Maybe converting the current and the initial story objects to JS strings and then simply comparing their equality.

Reply:
What do JS strings have to do with it?
Actually, I think we will need a more complicated way of determining if there are unsaved changes because of the audio recording feature that is to come. We need to only upload the audio when saving, and also keep track of replaced audio to delete it.
This should come naturally when implementing the audio recording feature, so we don't need to worry about it right now